sympy/functions/elementary/exponential.py

```diff
@@ -108,12 +108,12 @@ def _eval_power(self, other):
         return Pow._eval_power(Pow(b, e, evaluate=False), other)
 
     def _eval_expand_power_exp(self, **hints):
+        from sympy import Sum, Product
         arg = self.args[0]
         if arg.is_Add and arg.is_commutative:
-            expr = 1
-            for x in arg.args:
-                expr *= self.func(x)
-            return expr
+            return Mul.fromiter(self.func(x) for x in arg.args)
+        elif isinstance(arg, Sum) and arg.is_commutative:
+            return Product(exp(arg.function), *arg.limits)
```

Comment:
Should this be `self.func` rather than `exp`?

Also is this always valid in the case of infinite limits or are additional conditions needed?

Reply:
OK for `self.func`...

I don't know about infinite products.

Reply:
@smichr what do you think of infinite sums? Can they safely be expanded `exp(Sum(...)) ==> Product(exp(...), ...)`